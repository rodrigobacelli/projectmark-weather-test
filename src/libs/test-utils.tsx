import * as React from 'react';
import { render, RenderOptions } from '@testing-library/react';

const Providers = ({ children }: { children: React.ReactNode }) => {
  return children;
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render };
