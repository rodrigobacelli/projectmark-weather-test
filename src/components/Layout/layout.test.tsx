import { describe, it, expect } from 'vitest';
import { render } from '../../tests/test-utils';
import { Layout } from './';

describe('components/Layout', () => {
  it('should render the Container children', () => {
    const result = render(<Layout>Layout Content</Layout>);

    expect(result.getByText('Layout Content')).toBeInTheDocument();
  });

  it('should accept classNames', () => {
    const result = render(
      <Layout className="layout-className">Layout Content</Layout>
    );

    expect(
      result.container.querySelector('.layout-className')
    ).toBeInTheDocument();
  });

  it('should accept render the Header', () => {
    const result = render(<Layout>Layout Content</Layout>);

    expect(result.container.querySelector('header')).toBeInTheDocument();
  });

  it('should accept render the Footer', () => {
    const result = render(<Layout>Layout Content</Layout>);

    expect(result.container.querySelector('footer')).toBeInTheDocument();
  });

  it('should render the Layout Component', () => {
    const result = render(
      <Layout className="layout-className">Layout Content</Layout>
    );

    expect(result).toMatchSnapshot();
  });
});
