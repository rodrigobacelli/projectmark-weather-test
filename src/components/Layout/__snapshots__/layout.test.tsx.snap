// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Layout > should render the Layout Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen flex flex-col items-center bg-gray-100 dark:bg-gray-900 relative layout-className"
      >
        <header
          class="w-full bg-gray-200 dark:bg-gray-700 flex justify-center"
        >
          <div
            class="px-4 container py-4 flex items-center justify-between"
          >
            <div
              class="flex items-center gap-2"
            >
              <svg
                class="w-16 h-16"
                viewBox="0 0 500 500"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <style>
                    .cls-2{fill:#231815}
                  </style>
                </defs>
                <g>
                  <path
                    d="M215.44 305.85h23.82l5.43-5.2a7.65 7.65 0 0 1 10.61 0l5.43 5.2h23.82c0-3.75 2.42-7.5 7.25-7.5h28.36a81.5 81.5 0 0 0-54.33-121.91c-33.28-6.74-68.23 8.91-85.76 37.86a82.85 82.85 0 0 0-.52 84.06h28.65c4.8-.01 7.24 3.74 7.24 7.49Z"
                    fill="#f8b62d"
                  />
                  <path
                    class="cls-2"
                    d="M180.07 214.3c17.54-28.95 52.48-44.6 85.76-37.86a82.68 82.68 0 0 1 65.94 80.25c.07-9.66 15.07-9.65 15 0a98 98 0 0 0-64-91.07C245 151.8 200.82 164 175.45 195.05c-23.69 29-28.74 69.75-12.79 103.3h16.88a82.85 82.85 0 0 1 .53-84.05Z"
                  />
                  <path
                    class="cls-2"
                    d="M331.77 256.69c.07 9.65 15.07 9.67 15 0s-14.93-9.69-15 0Z"
                  />
                  <path
                    class="cls-2"
                    d="M331.77 256.69a83.48 83.48 0 0 1-11.6 41.66H337a97.63 97.63 0 0 0 9.78-41.66c.06 9.67-14.94 9.65-15.01 0ZM257.5 131.84V98.18c0-9.65-15-9.67-15 0v33.65c0 9.66 15 9.67 15 .01ZM250 98.18Zm-56 46.6-16.83-29.15c-4.83-8.36-17.79-.81-13 7.57L181 152.35c4.92 8.36 17.89.81 13-7.57Zm-6.48 3.79-16.83-29.15Zm-78.58 35.34 29.15 16.83c8.38 4.84 15.94-8.12 7.57-13L116.51 171c-8.37-4.87-15.94 8.08-7.57 12.91Zm3.79-6.48 29.15 16.83Zm-21.24 86.76h33.65c9.65 0 9.67-15 0-15H91.49c-9.65 0-9.67 15 0 15Zm0-7.5Zm317.02-7.5h-33.65c-9.65 0-9.67 15 0 15h33.65c9.65 0 9.67-15 0-15Zm-33.65 7.5Zm-12.95-55.95 29.15-16.83c8.36-4.83.81-17.79-7.57-13l-29.15 16.83c-8.34 4.87-.81 17.84 7.57 13Zm25.36-23.3-29.15 16.83Zm-68.37-25.09 16.83-29.15c4.84-8.37-8.12-15.94-13-7.57L306 144.78c-4.88 8.37 8.07 15.94 12.9 7.57Zm10.35-32.93-16.83 29.15Zm78.07 178.93H291.8c-4.83 0-7.24 3.75-7.25 7.5s2.41 7.5 7.25 7.5h115.52c9.68 0 9.68-15 0-15Zm-235.74 15h36.62c4.83 0 7.24-3.75 7.24-7.5s-2.41-7.5-7.24-7.5H92.68c-9.65 0-9.67 15 0 15h78.89Z"
                  />
                  <path
                    class="cls-2"
                    d="M375.15 338.46h-80.34l-8.71-8.33-25.37-24.28-5.43-5.2a7.65 7.65 0 0 0-10.61 0l-5.43 5.2-25.18 24.09-8.9 8.51h-80.33c-9.65 0-9.67 15 0 15h61.74c7.14 0 14.48.49 21.61 0 3.86-.26 6.59-3.27 9.22-5.92l1-.94.58-.59 10.72-10.26 1.77-1.69L250 316.36c6.65 6.5 13.25 13.12 20 19.61 3.5 3.38 7 6.73 10.63 10 .69.63 1.39 1.24 2.08 1.86 1.24 1.11 2.47 2.22 3.73 3.3 5 4.36 15.88 2.34 22 2.34h66.82c9.54-.01 9.56-15.01-.11-15.01Zm-28.65 40.11h-193c-9.65 0-9.67 15 0 15h193c9.65 0 9.67-15 0-15Z"
                  />
                </g>
              </svg>
              <b
                class="text-gray-900 dark:text-gray-100 text-2xl"
              >
                Weather App
              </b>
            </div>
            <div
              class="relative"
            >
              <div
                aria-controls="theme-dropdown-listbox"
                aria-expanded="false"
                aria-haspopup="listbox"
                class="hover:cursor-pointer rounded-full px-4 py-2 border border-gray-800 dark:border-gray-200 hover:bg-gray-800/20 hover:dark:bg-gray-200/20"
                id="theme-dropdown"
                role="combobox"
                tabindex="0"
              >
                <p
                  class="text-gray-900 dark:text-gray-100 flex gap-2 items-center text-sm"
                >
                  <svg
                    class="w-4 h-4"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 256c0-106-86-192-192-192l0 384c106 0 192-86 192-192zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
                    />
                  </svg>
                  <span
                    class="hidden sm:inline-flex"
                  >
                    System
                  </span>
                  <svg
                    class="ml-2 w-3 h-3"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                    />
                  </svg>
                </p>
              </div>
            </div>
          </div>
        </header>
        <div
          class="w-full flex flex-col items-center flex-1 pt-16 pb-10"
        >
          Layout Content
        </div>
        <footer
          class="w-full flex flex-col items-center justify-center bg-gray-200 dark:bg-gray-800"
        >
          <div
            class="container px-4 py-8"
          >
            <div
              class="flex items-center gap-2"
            >
              <svg
                class="w-12 h-12"
                viewBox="0 0 500 500"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <style>
                    .cls-2{fill:#231815}
                  </style>
                </defs>
                <g>
                  <path
                    d="M215.44 305.85h23.82l5.43-5.2a7.65 7.65 0 0 1 10.61 0l5.43 5.2h23.82c0-3.75 2.42-7.5 7.25-7.5h28.36a81.5 81.5 0 0 0-54.33-121.91c-33.28-6.74-68.23 8.91-85.76 37.86a82.85 82.85 0 0 0-.52 84.06h28.65c4.8-.01 7.24 3.74 7.24 7.49Z"
                    fill="#f8b62d"
                  />
                  <path
                    class="cls-2"
                    d="M180.07 214.3c17.54-28.95 52.48-44.6 85.76-37.86a82.68 82.68 0 0 1 65.94 80.25c.07-9.66 15.07-9.65 15 0a98 98 0 0 0-64-91.07C245 151.8 200.82 164 175.45 195.05c-23.69 29-28.74 69.75-12.79 103.3h16.88a82.85 82.85 0 0 1 .53-84.05Z"
                  />
                  <path
                    class="cls-2"
                    d="M331.77 256.69c.07 9.65 15.07 9.67 15 0s-14.93-9.69-15 0Z"
                  />
                  <path
                    class="cls-2"
                    d="M331.77 256.69a83.48 83.48 0 0 1-11.6 41.66H337a97.63 97.63 0 0 0 9.78-41.66c.06 9.67-14.94 9.65-15.01 0ZM257.5 131.84V98.18c0-9.65-15-9.67-15 0v33.65c0 9.66 15 9.67 15 .01ZM250 98.18Zm-56 46.6-16.83-29.15c-4.83-8.36-17.79-.81-13 7.57L181 152.35c4.92 8.36 17.89.81 13-7.57Zm-6.48 3.79-16.83-29.15Zm-78.58 35.34 29.15 16.83c8.38 4.84 15.94-8.12 7.57-13L116.51 171c-8.37-4.87-15.94 8.08-7.57 12.91Zm3.79-6.48 29.15 16.83Zm-21.24 86.76h33.65c9.65 0 9.67-15 0-15H91.49c-9.65 0-9.67 15 0 15Zm0-7.5Zm317.02-7.5h-33.65c-9.65 0-9.67 15 0 15h33.65c9.65 0 9.67-15 0-15Zm-33.65 7.5Zm-12.95-55.95 29.15-16.83c8.36-4.83.81-17.79-7.57-13l-29.15 16.83c-8.34 4.87-.81 17.84 7.57 13Zm25.36-23.3-29.15 16.83Zm-68.37-25.09 16.83-29.15c4.84-8.37-8.12-15.94-13-7.57L306 144.78c-4.88 8.37 8.07 15.94 12.9 7.57Zm10.35-32.93-16.83 29.15Zm78.07 178.93H291.8c-4.83 0-7.24 3.75-7.25 7.5s2.41 7.5 7.25 7.5h115.52c9.68 0 9.68-15 0-15Zm-235.74 15h36.62c4.83 0 7.24-3.75 7.24-7.5s-2.41-7.5-7.24-7.5H92.68c-9.65 0-9.67 15 0 15h78.89Z"
                  />
                  <path
                    class="cls-2"
                    d="M375.15 338.46h-80.34l-8.71-8.33-25.37-24.28-5.43-5.2a7.65 7.65 0 0 0-10.61 0l-5.43 5.2-25.18 24.09-8.9 8.51h-80.33c-9.65 0-9.67 15 0 15h61.74c7.14 0 14.48.49 21.61 0 3.86-.26 6.59-3.27 9.22-5.92l1-.94.58-.59 10.72-10.26 1.77-1.69L250 316.36c6.65 6.5 13.25 13.12 20 19.61 3.5 3.38 7 6.73 10.63 10 .69.63 1.39 1.24 2.08 1.86 1.24 1.11 2.47 2.22 3.73 3.3 5 4.36 15.88 2.34 22 2.34h66.82c9.54-.01 9.56-15.01-.11-15.01Zm-28.65 40.11h-193c-9.65 0-9.67 15 0 15h193c9.65 0 9.67-15 0-15Z"
                  />
                </g>
              </svg>
              <b
                class="text-gray-900 dark:text-gray-100 text-lg"
              >
                Weather App
              </b>
            </div>
          </div>
          <div
            class="flex w-full justify-center py-2 bg-gray-300 dark:bg-gray-700"
          >
            <p
              class="text-gray-900 dark:text-gray-100"
            >
              Weather App @ 2025
            </p>
          </div>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen flex flex-col items-center bg-gray-100 dark:bg-gray-900 relative layout-className"
    >
      <header
        class="w-full bg-gray-200 dark:bg-gray-700 flex justify-center"
      >
        <div
          class="px-4 container py-4 flex items-center justify-between"
        >
          <div
            class="flex items-center gap-2"
          >
            <svg
              class="w-16 h-16"
              viewBox="0 0 500 500"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <style>
                  .cls-2{fill:#231815}
                </style>
              </defs>
              <g>
                <path
                  d="M215.44 305.85h23.82l5.43-5.2a7.65 7.65 0 0 1 10.61 0l5.43 5.2h23.82c0-3.75 2.42-7.5 7.25-7.5h28.36a81.5 81.5 0 0 0-54.33-121.91c-33.28-6.74-68.23 8.91-85.76 37.86a82.85 82.85 0 0 0-.52 84.06h28.65c4.8-.01 7.24 3.74 7.24 7.49Z"
                  fill="#f8b62d"
                />
                <path
                  class="cls-2"
                  d="M180.07 214.3c17.54-28.95 52.48-44.6 85.76-37.86a82.68 82.68 0 0 1 65.94 80.25c.07-9.66 15.07-9.65 15 0a98 98 0 0 0-64-91.07C245 151.8 200.82 164 175.45 195.05c-23.69 29-28.74 69.75-12.79 103.3h16.88a82.85 82.85 0 0 1 .53-84.05Z"
                />
                <path
                  class="cls-2"
                  d="M331.77 256.69c.07 9.65 15.07 9.67 15 0s-14.93-9.69-15 0Z"
                />
                <path
                  class="cls-2"
                  d="M331.77 256.69a83.48 83.48 0 0 1-11.6 41.66H337a97.63 97.63 0 0 0 9.78-41.66c.06 9.67-14.94 9.65-15.01 0ZM257.5 131.84V98.18c0-9.65-15-9.67-15 0v33.65c0 9.66 15 9.67 15 .01ZM250 98.18Zm-56 46.6-16.83-29.15c-4.83-8.36-17.79-.81-13 7.57L181 152.35c4.92 8.36 17.89.81 13-7.57Zm-6.48 3.79-16.83-29.15Zm-78.58 35.34 29.15 16.83c8.38 4.84 15.94-8.12 7.57-13L116.51 171c-8.37-4.87-15.94 8.08-7.57 12.91Zm3.79-6.48 29.15 16.83Zm-21.24 86.76h33.65c9.65 0 9.67-15 0-15H91.49c-9.65 0-9.67 15 0 15Zm0-7.5Zm317.02-7.5h-33.65c-9.65 0-9.67 15 0 15h33.65c9.65 0 9.67-15 0-15Zm-33.65 7.5Zm-12.95-55.95 29.15-16.83c8.36-4.83.81-17.79-7.57-13l-29.15 16.83c-8.34 4.87-.81 17.84 7.57 13Zm25.36-23.3-29.15 16.83Zm-68.37-25.09 16.83-29.15c4.84-8.37-8.12-15.94-13-7.57L306 144.78c-4.88 8.37 8.07 15.94 12.9 7.57Zm10.35-32.93-16.83 29.15Zm78.07 178.93H291.8c-4.83 0-7.24 3.75-7.25 7.5s2.41 7.5 7.25 7.5h115.52c9.68 0 9.68-15 0-15Zm-235.74 15h36.62c4.83 0 7.24-3.75 7.24-7.5s-2.41-7.5-7.24-7.5H92.68c-9.65 0-9.67 15 0 15h78.89Z"
                />
                <path
                  class="cls-2"
                  d="M375.15 338.46h-80.34l-8.71-8.33-25.37-24.28-5.43-5.2a7.65 7.65 0 0 0-10.61 0l-5.43 5.2-25.18 24.09-8.9 8.51h-80.33c-9.65 0-9.67 15 0 15h61.74c7.14 0 14.48.49 21.61 0 3.86-.26 6.59-3.27 9.22-5.92l1-.94.58-.59 10.72-10.26 1.77-1.69L250 316.36c6.65 6.5 13.25 13.12 20 19.61 3.5 3.38 7 6.73 10.63 10 .69.63 1.39 1.24 2.08 1.86 1.24 1.11 2.47 2.22 3.73 3.3 5 4.36 15.88 2.34 22 2.34h66.82c9.54-.01 9.56-15.01-.11-15.01Zm-28.65 40.11h-193c-9.65 0-9.67 15 0 15h193c9.65 0 9.67-15 0-15Z"
                />
              </g>
            </svg>
            <b
              class="text-gray-900 dark:text-gray-100 text-2xl"
            >
              Weather App
            </b>
          </div>
          <div
            class="relative"
          >
            <div
              aria-controls="theme-dropdown-listbox"
              aria-expanded="false"
              aria-haspopup="listbox"
              class="hover:cursor-pointer rounded-full px-4 py-2 border border-gray-800 dark:border-gray-200 hover:bg-gray-800/20 hover:dark:bg-gray-200/20"
              id="theme-dropdown"
              role="combobox"
              tabindex="0"
            >
              <p
                class="text-gray-900 dark:text-gray-100 flex gap-2 items-center text-sm"
              >
                <svg
                  class="w-4 h-4"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M448 256c0-106-86-192-192-192l0 384c106 0 192-86 192-192zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256z"
                  />
                </svg>
                <span
                  class="hidden sm:inline-flex"
                >
                  System
                </span>
                <svg
                  class="ml-2 w-3 h-3"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
                  />
                </svg>
              </p>
            </div>
          </div>
        </div>
      </header>
      <div
        class="w-full flex flex-col items-center flex-1 pt-16 pb-10"
      >
        Layout Content
      </div>
      <footer
        class="w-full flex flex-col items-center justify-center bg-gray-200 dark:bg-gray-800"
      >
        <div
          class="container px-4 py-8"
        >
          <div
            class="flex items-center gap-2"
          >
            <svg
              class="w-12 h-12"
              viewBox="0 0 500 500"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <style>
                  .cls-2{fill:#231815}
                </style>
              </defs>
              <g>
                <path
                  d="M215.44 305.85h23.82l5.43-5.2a7.65 7.65 0 0 1 10.61 0l5.43 5.2h23.82c0-3.75 2.42-7.5 7.25-7.5h28.36a81.5 81.5 0 0 0-54.33-121.91c-33.28-6.74-68.23 8.91-85.76 37.86a82.85 82.85 0 0 0-.52 84.06h28.65c4.8-.01 7.24 3.74 7.24 7.49Z"
                  fill="#f8b62d"
                />
                <path
                  class="cls-2"
                  d="M180.07 214.3c17.54-28.95 52.48-44.6 85.76-37.86a82.68 82.68 0 0 1 65.94 80.25c.07-9.66 15.07-9.65 15 0a98 98 0 0 0-64-91.07C245 151.8 200.82 164 175.45 195.05c-23.69 29-28.74 69.75-12.79 103.3h16.88a82.85 82.85 0 0 1 .53-84.05Z"
                />
                <path
                  class="cls-2"
                  d="M331.77 256.69c.07 9.65 15.07 9.67 15 0s-14.93-9.69-15 0Z"
                />
                <path
                  class="cls-2"
                  d="M331.77 256.69a83.48 83.48 0 0 1-11.6 41.66H337a97.63 97.63 0 0 0 9.78-41.66c.06 9.67-14.94 9.65-15.01 0ZM257.5 131.84V98.18c0-9.65-15-9.67-15 0v33.65c0 9.66 15 9.67 15 .01ZM250 98.18Zm-56 46.6-16.83-29.15c-4.83-8.36-17.79-.81-13 7.57L181 152.35c4.92 8.36 17.89.81 13-7.57Zm-6.48 3.79-16.83-29.15Zm-78.58 35.34 29.15 16.83c8.38 4.84 15.94-8.12 7.57-13L116.51 171c-8.37-4.87-15.94 8.08-7.57 12.91Zm3.79-6.48 29.15 16.83Zm-21.24 86.76h33.65c9.65 0 9.67-15 0-15H91.49c-9.65 0-9.67 15 0 15Zm0-7.5Zm317.02-7.5h-33.65c-9.65 0-9.67 15 0 15h33.65c9.65 0 9.67-15 0-15Zm-33.65 7.5Zm-12.95-55.95 29.15-16.83c8.36-4.83.81-17.79-7.57-13l-29.15 16.83c-8.34 4.87-.81 17.84 7.57 13Zm25.36-23.3-29.15 16.83Zm-68.37-25.09 16.83-29.15c4.84-8.37-8.12-15.94-13-7.57L306 144.78c-4.88 8.37 8.07 15.94 12.9 7.57Zm10.35-32.93-16.83 29.15Zm78.07 178.93H291.8c-4.83 0-7.24 3.75-7.25 7.5s2.41 7.5 7.25 7.5h115.52c9.68 0 9.68-15 0-15Zm-235.74 15h36.62c4.83 0 7.24-3.75 7.24-7.5s-2.41-7.5-7.24-7.5H92.68c-9.65 0-9.67 15 0 15h78.89Z"
                />
                <path
                  class="cls-2"
                  d="M375.15 338.46h-80.34l-8.71-8.33-25.37-24.28-5.43-5.2a7.65 7.65 0 0 0-10.61 0l-5.43 5.2-25.18 24.09-8.9 8.51h-80.33c-9.65 0-9.67 15 0 15h61.74c7.14 0 14.48.49 21.61 0 3.86-.26 6.59-3.27 9.22-5.92l1-.94.58-.59 10.72-10.26 1.77-1.69L250 316.36c6.65 6.5 13.25 13.12 20 19.61 3.5 3.38 7 6.73 10.63 10 .69.63 1.39 1.24 2.08 1.86 1.24 1.11 2.47 2.22 3.73 3.3 5 4.36 15.88 2.34 22 2.34h66.82c9.54-.01 9.56-15.01-.11-15.01Zm-28.65 40.11h-193c-9.65 0-9.67 15 0 15h193c9.65 0 9.67-15 0-15Z"
                />
              </g>
            </svg>
            <b
              class="text-gray-900 dark:text-gray-100 text-lg"
            >
              Weather App
            </b>
          </div>
        </div>
        <div
          class="flex w-full justify-center py-2 bg-gray-300 dark:bg-gray-700"
        >
          <p
            class="text-gray-900 dark:text-gray-100"
          >
            Weather App @ 2025
          </p>
        </div>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
